massa-final-state/src/final_state.rs

```diff
@@ -117,15 +118,37 @@ impl FinalState {
             self.changes_history.push_back((slot, changes));
         }
 
+        // final hash computing and sub hashes logging
+        // 1. init hash concatenation with the ledger hash
+        let ledger_hash = self.ledger.get_ledger_hash();
+        let mut hash_concat: Vec<u8> = ledger_hash.to_bytes().to_vec();
+        debug!("ledger hash at slot {}: {}", slot, ledger_hash);
+        // 2. async_pool hash
+        hash_concat.extend(self.async_pool.hash.to_bytes());
+        debug!("async_pool hash at slot {}: {}", slot, self.async_pool.hash);
+        // 3. pos deferred_credit hash
+        hash_concat.extend(self.pos_state.deferred_credits.hash.to_bytes());
         debug!(
-            "ledger hash at slot {}: {}",
-            slot,
-            self.ledger.get_ledger_hash()
+            "deferred_credit hash at slot {}: {}",
+            slot, self.pos_state.deferred_credits.hash
         );
+        // 4. pos cycle history hashes
+        for cycle_info in &self.pos_state.cycle_history {
+            hash_concat.extend(cycle_info.hash.to_bytes());
+            debug!(
+                "cycle ({}) hash at slot {}: {}",
+                cycle_info.cycle, slot, cycle_info.hash
+            );
+        }
+        // 5. executed operations hash
+        hash_concat.extend(self.executed_ops.hash.to_bytes());
         debug!(
-            "executed_ops hash at slot {}: {:?}",
+            "executed_ops hash at slot {}: {}",
             slot, self.executed_ops.hash
         );
+        // 6. final state hash
+        let final_state_hash = Hash::compute_from(&hash_concat);
+        debug!("final_state hash at slot {}: {}", slot, final_state_hash);
```

Comment:
Can you make this one as "info" ? It will help us a lot to debug the peoples and our nodes that often run in info.

Reply:
Done